screens/dashboard/components/upcoming.js

```diff
@@ -0,0 +1,157 @@
+// @flow
+
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import {
+  StyleSheet,
+  Text,
+  View,
+  TouchableHighlight,
+} from 'react-native';
+import { connect } from 'react-redux';
+import Icon from 'react-native-vector-icons/FontAwesome';
+import moment from 'moment';
+
+// import global actions
+import * as actions from '../actions';
+
+// import global styles
+// @TODO: move the global styles from this screen into ../../styles/common 
+import commonStyles from '../../../styles/common';
+const styles = StyleSheet.create(commonStyles);
+
+type Props = {
+  actions: Object,
+  profile: Object,
+  navigation: Object
+};
+
+
+export class Upcoming extends Component<Props> {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      expanded: false
+    };
+    this.icons = {
+      'open': 'angle-down',
+      'close': 'angle-up'
+    };
+  }
+
+
+  showUpcoming(firstOrRemaining) {
```

Comment:
the ShowUpcoming method does not seem to be bound in the constructor. All methods that use the `this` keyword should be bound like this:

```
constructor(props){
super(props);
this.showUpcoming = this.showUpcoming.bind(this):
}```

Reply:
Moved to render()